docs/docs/guides/gitlab.mdx

```diff
@@ -0,0 +1,427 @@
+---
+id: gitlab
+title: Gitlab Hydra integration
+---
+
+Gitlab has several OAuth2 related features. The relevant here is the possibility
+to sign into GitLab with (almost) any OAuth2 provider, in this case ORY Hydra. The documentation for this
+is kept generic on purpose. So, in this guide, we'll connect
+GitLab's omniauth-connector to ORY Hydra. We'll do that in a docker-based
+lab-environment in order to investigate the details before you do something like
+this in production.
+
+## Preparation
+
+Even though we're mostly using hydra in a docker-container, having the
+command-line-client available is quite useful. So let's install hydra as
+explained in the [installation-guide](../install).
+You'll also need [docker](https://docs.docker.com/get-docker/) and
+[docker-compose](https://docs.docker.com/compose/install/).
+
+The [hydra-5-min-tutorial](../5min-tutorial) might
+be worth checking out upfront. It'll a give a nice quick overview how OAuth2 is
+working within hydra with a minimal example. We assume basic knowledge, here.
+
+If you have not yet the source code of hydra, which we'll need for the `docker-compose`
+yaml-files, clone this repository:
+
+```shell
+git clone git@github.com:ory/hydra.git
+```
+
+## Spin up the instances
+
+Gitlab has some [documentation](https://docs.gitlab.com/omnibus/docker/) about
+how to use their docker-images. It has also an example for docker-compose. So change into the directory with the Hydra 
+source code and create a new file quickstart-gitlab.yml with this content:
+
+```yaml
+version: '3'
+
+services:
+  gitlab:
+    image: gitlab/gitlab-ce:latest
+    restart: always
+    hostname: gitlab.example.com
+    environment:
+      GITLAB_OMNIBUS_CONFIG: |
+        external_url 'http://gitlab.example.com:8000/'
+    ports:
+      - "443:443" # https
+      - "8000:8000" # http
+      - "2222:22" # ssh
+    volumes:
+    - '$GITLAB_HOME/config:/etc/gitlab'
+    - '$GITLAB_HOME/logs:/var/log/gitlab'
+    - '$GITLAB_HOME/data:/var/opt/gitlab'
```

Comment:
Given that `$GITLAB_HOME` is set to `GITLAB_HOME=$(pwd)/gitlab` we can probably reduce errors (e.g. on Windows) by just using `./gitlab/config:/etc/gitlab`